File: Kalorie/Kalorie/Core/Models/FoodItemReportModel.swift
//
//  FoodItemReportModel.swift
//  Kalorie
//
//  Created by Josef Antoni on 17.09.2026.
//

import Foundation

struct FoodItemReportDomain: Equatable {

    // MARK: - Properties

    let barcode: String
    let reportedBy: String
    let reason: String
    let reportedAt: Date

    // MARK: - Functions

    static func id(barcode: String, userId: String) -> String {
        "\(barcode)_\(userId)"
    }
}

enum FoodItemReportError: Error {
    case reasonRequired
    case reasonTooLong
}

File: Android/app/src/main/java/antoni/kalorie/core/models/FoodItemReportModel.kt
package antoni.kalorie.core.models

import java.time.Instant

data class FoodItemReportDomain(
    val barcode: String,
    val reportedBy: String,
    val reason: String,
    val reportedAt: Instant,
) {

    // MARK: - Functions

    companion object {
        fun id(barcode: String, userId: String): String = "${barcode}_$userId"
    }
}

sealed class FoodItemReportError : Exception() {
    data object ReasonRequired : FoodItemReportError()
    data object ReasonTooLong : FoodItemReportError()
}
